app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -157,32 +158,42 @@ func (ctw CryptoTransferWatcher) beginWatching(q *queue.Queue) {
 func (ctw CryptoTransferWatcher) processTransaction(tx transaction.HederaTransaction, q *queue.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
 
-	var amount int64
+	var amount string
 	for _, tr := range tx.Transfers {
 		if tr.Account == ctw.accountID.String() {
-			amount = tr.Amount
+			amount = fmt.Sprint(tr.Amount)
 			break
 		}
 	}
 
+	wholeMemoCheck := regexp.MustCompile("^0x([A-Fa-f0-9]){40}-[1-9][0-9]+-[1-9][0-9]+$")
+
 	decodedMemo, e := base64.StdEncoding.DecodeString(tx.MemoBase64)
-	if e != nil || len(decodedMemo) < 42 {
+	if e != nil || len(decodedMemo) < 42 || !wholeMemoCheck.MatchString(string(decodedMemo)) {
 		ctw.logger.Errorf("Could not verify transaction memo - Error: [%s]", e)
 		return
 	}
 
-	ethAddress := decodedMemo[:42]
-	feeString := string(decodedMemo[42:])
+	memo := strings.Split(string(decodedMemo), "-")
+	ethAddress := memo[0]
+	fee := memo[1]
+	gasPrice := memo[2] // In gwei
 
 	re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")
-	if !re.MatchString(string(ethAddress)) {
+	if !re.MatchString(ethAddress) {
 		ctw.logger.Errorf("Could not verify Ethereum Address - [%s]", ethAddress)
 		return
 	}
 
-	_, e = helper.ToBigInt(feeString)
+	_, e = helper.ToBigInt(fee)
+	if e != nil {
+		ctw.logger.Errorf("Could not verify transaction fee - [%s]", fee)
```

Comment:
`verify` is not the correct word. Better use `parse`.
We could add the ID of the Transaction as-well to have better audit trail afterwards.